providers/implementations/storemgmt/file_store_any2obj.c

```diff
@@ -45,33 +46,76 @@
 static OSSL_FUNC_decoder_newctx_fn any2obj_newctx;
 static OSSL_FUNC_decoder_freectx_fn any2obj_freectx;
 
+struct any2obj_ctx_st {
+    PROV_CTX *provctx;
+    char data_structure[OSSL_MAX_CODEC_STRUCT_SIZE];
+};
+
 static void *any2obj_newctx(void *provctx)
 {
-    return provctx;
+    struct any2obj_ctx_st *ctx = OPENSSL_zalloc(sizeof(*ctx));
+
+    if (ctx != NULL)
+        ctx->provctx = provctx;
+    return ctx;
 }
 
-static void any2obj_freectx(void *vctx)
+static void any2obj_freectx(void *ctx)
 {
+    OPENSSL_free(ctx);
 }
 
-static int any2obj_decode_final(void *provctx, int objtype, BUF_MEM *mem,
+static int any2obj_set_ctx_params(void *vctx, const OSSL_PARAM params[])
+{
+    struct any2obj_ctx_st *ctx = vctx;
+    const OSSL_PARAM *p;
+    char *str;
```

Comment:
```suggestion
    char *str = NULL;
```
Please note that it is always a good practice to initialize the variables.

Reply:
No, it is not always a good practice to combine declaration with initialisation.
Sometimes this masks failure to *correctly* initialise a variable that would be highlighted by compiler warnings.